src/index.ts

```diff
@@ -21,7 +21,8 @@ export function identify(query: string, options: IdentifyOptions = {}): Identify
     throw new Error(`Unknown dialect. Allowed values: ${DIALECTS.join(', ')}`);
   }
 
-  const result = parse(query, isStrict, dialect, options.identifyTables);
+  const result = parse(query, isStrict, dialect, options.identifyTables, options.paramTypes);
```

Comment:
I think that instead of passing `options.paramTypes` undefined through if not set, we should set default values that are adapter specific, so that all adapters go through the same logic of parsing parameters, vs having both the `scanCustomParameter` function as well as adapter specific handlers below.

So like:
```
if (!options.paramTypes) {
  if (dialect === 'sqlite') {
    options.paramTypes = { positional: true };
  }
  elseif (dialect === 'psql') {
    options.paramTypes = { numbered: [ '$' ] };
  }
  // other options...
}
```

We could also do this transformation in the `tokenizer` as well, I've no particularly strong preference, just think we should do this instead of having both code paths.

Reply:
Great idea, will do